src/Npgsql/Util/StringBuilderExtensions.cs

```diff
@@ -10,13 +10,29 @@ namespace Npgsql.Util
     internal static class StringBuilderExtensions
     {
         /// <summary>
-        /// Appends the provided <see cref="ReadOnlySpan{T}"/> to the <see cref="StringBuilder"/> by calling ToString on
-        /// the span.
+        /// Appends the provided <see cref="ReadOnlySpan{T}"/> to the <see cref="StringBuilder"/>.
         /// </summary>
         /// <param name="stringBuilder">The <see cref="StringBuilder"/> to append to.</param>
         /// <param name="span">The <see cref="ReadOnlySpan{T}"/> to append.</param>
-        public static void Append(this StringBuilder stringBuilder, ReadOnlySpan<char> span)
-            => stringBuilder.Append(span.ToString());
+        public static StringBuilder Append(this StringBuilder stringBuilder, ReadOnlySpan<char> span)
+        {
+#if NET452
+            return stringBuilder.Append(span.ToString());
+#else
+            if (span.Length > 0)
+            {
+                unsafe
+                {
+                    fixed (char* value = &span.GetPinnableReference())
+                    {
+                        return stringBuilder.Append(value, span.Length);
+                    }
+                }
+            }
```

Comment:
Is there any way to reduce brace number? Otherwise, looks good.

Reply:
The braces for the `if` and `fixed` statements can be omitted, but not for `unsafe`, so it would look really weird, IMO:

```csharp
if (span.Length > 0)
    unsafe
    {
        fixed (char* value = &span.GetPinnableReference())
            return stringBuilder.Append(value, span.Length);
    }
```

Reply:
Hm, not usual, but fine to me (:

Reply:
For me too.

Reply:
I'd just leave it as-is with the curlies...